src/python_testing/matter_testing_infrastructure/chip/testing/matter_stack_state.py

```diff
@@ -0,0 +1,102 @@
+#
+#    Copyright (c) 2025 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import builtins
+import logging
+from typing import TYPE_CHECKING
+
+import chip
+from chip.ChipStack import ChipStack
+from chip.storage import PersistentStorage
+
+logger = logging.getLogger("matter.python_testing")
+logger.setLevel(logging.INFO)
+
+if TYPE_CHECKING:
+    from chip.testing.matter_testing import MatterTestConfig
+
+
+class MatterStackState:
```

Comment:
since we touch this, could you add docstrings to this class and module describing a little bit why it is needed?